FreeRTOS/FreeRTOS-Plus-TCP
# STM32Hxx porting fixes and suggestions

STM32Hxx porting fixes and suggestions

Description
-----------
The changes are separated in different changesets so you may choose which one you may like to pick (if any). The first one is the most straightforward since it fixes code using functions that don't exist (they might have been renamed and the code was not updated).

The other changesets are suggestions on how to avoid conflicts with auto-generated files from  STM32CubeIDE's project and original HAL ETH driver. Some suggestions may seem hacky but they ease maintenance allowing the ST's code generator to be used for reconfiguring the project without losing changes or reintroducing the conflicts.

Test Steps
-----------
The original code did not compiled. 

Checklist:
----------
- [x] I have tested my changes. No regression in existing tests. NOTE: I have just brought the interface up and tested with ping (it works). 

Related Issue
-----------
None.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
